dvc/output/base.py

```diff
@@ -149,7 +149,11 @@ def cache_path(self):
 
     @property
     def checksum(self):
-        return self.info.get(self.remote.PARAM_CHECKSUM)
+        # TODO it will always call get, need to write it somewhere
+        checksum = self.info.get(self.remote.PARAM_CHECKSUM)
+        if not checksum:
+            checksum = self.remote.get_checksum(self.path_info)
```

Comment:
This property should only return what is already computed, it shouldn't start computing it all over again.

See dvc/outpt/base.py:get_used_cache, where the issue rose. We check `if not self.info` there, while we should probably check `if not self.checksum`. Push shouldn't push uncommited changes, as it doesn't even have the cache for them in .dvc/cache.